File: python/app/models/event.py
from sqlalchemy import Column, BigInteger, String, DateTime, ForeignKey, Float
from sqlalchemy.orm import relationship
from app.core.database import Base
import datetime

class Event(Base):
    __tablename__ = "events"
    
    id = Column(BigInteger, primary_key=True, index=True)
    test_id = Column(String, ForeignKey("tests.id"), index=True, nullable=False)
    user_id = Column(String, index=True, nullable=False)  # ID пользователя на клиенте
    variant = Column(String, nullable=False)  # 'A' или 'B'
    event_type = Column(String, nullable=False)  # impression, click, conversion
    value = Column(Float, default=1.0)
    created_at = Column(DateTime, default=datetime.datetime.utcnow, index=True)
    
    test = relationship("Test", back_populates="events")

File: go/internal/model/event.go
package model

import "time"

// Event represents an event from the client (impression, click, conversion)
type Event struct {
	ID        int64     `json:"id"`
	TestID    string    `json:"test_id"`
	UserID    string    `json:"user_id"`
	Variant   string    `json:"variant"`    // 'A' or 'B'
	Type      string    `json:"event_type"` // 'impression', 'click', 'conversion'
	Value     float64   `json:"value"`      // weight of the event (default 1.0)
	CreatedAt time.Time `json:"created_at"`
}
